packages/AlgernonTests.package/ALGTestGeneral.class/instance/testAdvancedChildNavigation.st

```diff
@@ -0,0 +1,11 @@
+testing
+testAdvancedChildNavigation
+	| oldQuery |
+	self runQuery: 'AlgernonTests'.
+	self moveCursorTo: 'Category'.
+	oldQuery := self currentQuery.
+	self algerInstance showChildList.
+	self algerInstance showChildList. "Navigate all the way in"
+	self algerInstance showParentList.
+	self algerInstance showParentList.
+	self assert: (oldQuery = (self currentQuery))
```

Comment:
Why would this trigger the size to be 0?

Reply:
Because the search term is deleted when the children are accessed; thus has the length 0.
(I made the test a bit more general, because we don't have a test for viewing children more than one time)

Reply:
Why is the search term even deleted when we view children? Seems like a weird behavior to me.

Reply:
I think we should discuss this with the client. Could you write an additional, lower-level test that clears the term "manually" to be future proof should this behavior change?

Reply:
Because you should be able to search trough the Children.
(And the original term is restored, when you navigate to the parent)

Reply:
Ah! Now I see. That's non-obvious. Maybe we should have an indicator.